hare/hare_rounds_test.go

```diff
@@ -124,7 +124,9 @@ func Test_HarePreRoundEmptySet(t *testing.T) {
 	var mu sync.RWMutex
 	m := [layers][nodes]int{}
 
-	w := runNodesFor(t, nodes, 2, layers, 2, false,
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
You can add `defer cancel()` here. Calling cancel after the first time is a no-op. Without the defer, if the test fails between here and line 147 we have a dead go routine.